### PR TITLE
Simple calendar integration

### PR DESCRIPTION
**Pull request by tortuetorche**
polls_calendar_new_event
References:
#121, #136 and #411

### CODE REVIEW

File: src/js/components/VoteTable/VoteCalendarLink.vue [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+			const dateTimeStart = option.timestamp
+			const dateTimeEnd = moment.unix(dateTimeStart).add(1, 'hour').unix()
+
+			const protocol = window.location.protocol + '//'
+			const host = window.location.host
+			// Useful if the current pathname is '/index.php/apps/...'
+			const pathname = window.location.pathname.match('^(.*?/apps)/.*$')
+			const appsRoot = pathname ? pathname[1] : '/apps'
+
+			const calendarRoot = appsRoot + '/calendar'
```
Comment (v1r0x) [inline src/js/components/VoteTable/VoteCalendarLink.vue]: NC already has a function to get the URL of an app, so you don't have to fiddle with `window.location`. It's simply `OC.generateUrl('calendar');` :)

File: src/js/components/VoteTable/VoteTableHeader.vue [hunk 2]
```diff
@@ -41,6 +41,17 @@
 			</div>
 		</div>
 
+		<div v-if="acl.allowEdit && poll.type === 'datePoll'" class="new-event">
+			<a class="button"
+				:class="{
+					primary: isWinner,
+					'icon-calendar': isWinner,
+					'icon-calendar-000': !isWinner
```
Comment (dartcafe) [inline src/js/components/VoteTable/VoteTableHeader.vue]: You could put this into the component as template, add some props  and make it reusable.

File: src/js/components/VoteTable/VoteList.vue [hunk 3]
```diff
@@ -31,6 +31,12 @@
 			<PollItemText v-if="poll.type === 'textPoll'" :option="option" />
 			<PollItemDate v-if="poll.type === 'datePoll'" :option="option" />
 
+			<div v-if="acl.allowEdit && poll.type === 'datePoll'" class="new-event">
+				<a class="button icon-calendar-000"
+					:title="t('polls', 'Add a calendar event')"
```
Comment (dartcafe) [inline src/js/components/VoteTable/VoteList.vue]: ... and replace this with the component too.

File: lib/Model/Acl.php [hunk 4]
```diff
@@ -316,6 +323,18 @@ public function getAllowEdit(): bool {
...
+	 * @return bool
+	 */
+	public function getAllowAddEvent(): bool {
+		return (
+			   $this->getAllowEdit()
+			&& $this->poll->getType() === 'datePoll'
+			&& $this->appManager->isEnabledForUser('calendar')
+		);
+	}
+
```
Comment (dartcafe) [inline lib/Model/Acl.php]: @tortuetorche I added a new check to the acl: getAllowAddEvent().
It checks the access to the calendar app and integrates your condition.

**Comment by dartcafe** [comment]
@tortuetorche Are you planning to continue this?

**Comment by tortuetorche** [comment]
Hi @dartcafe

Sorry, not in the next weeks...
If anybody want to continue or borrow this pull request, I'll be very grateful!

Have a good day,
Tortue Torche

**Comment by dartcafe** [comment]
I think, too much has changed in the core of polls. At least, with https://github.com/nextcloud/server/issues/20154 a new approach makes sense, since we can do this in the backend. 

I would consider closing this and wait for https://github.com/nextcloud/server/issues/20154

**Pull request closed without merging**